### ISSUE TITLE
Cilium didn't update CEP IP address when a pod restarted

### ISSUE THREAD

**Issue by YesemKebede** [cilium/cilium#14406]
## Bug report

We have cilium polices for some of our pods and we started getting socket timeout errors and looking in to it, we saw the following drops 
```
kubectl -n kube-system exec -ti cilium-8rdzn -- cilium monitor --type drop
Listening for events on 4 CPUs with 64x4096 of shared memory
Press Ctrl-C to quit
level=info msg="Initializing dissection cache..." subsys=monitor
xx drop (Policy denied) flow 0x62ed62ed to endpoint 1158, identity 2->47976: 10.20.146.233:51630 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x62ed62ed to endpoint 1158, identity 2->47976: 10.20.146.233:51630 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x62ed62ed to endpoint 1158, identity 2->47976: 10.20.146.233:51630 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x62ed62ed to endpoint 1158, identity 2->47976: 10.20.146.233:51630 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x62ed62ed to endpoint 1158, identity 2->47976: 10.20.146.233:51630 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x62ed62ed to endpoint 1158, identity 2->47976: 10.20.146.233:51630 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x62ed62ed to endpoint 1158, identity 2->47976: 10.20.146.233:51630 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x245f245f to endpoint 1158, identity 2->47976: 10.20.146.233:34060 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x245f245f to endpoint 1158, identity 2->47976: 10.20.146.233:34060 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x245f245f to endpoint 1158, identity 2->47976: 10.20.146.233:34060 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x245f245f to endpoint 1158, identity 2->47976: 10.20.146.233:34060 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x245f245f to endpoint 1158, identity 2->47976: 10.20.146.233:34060 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x245f245f to endpoint 1158, identity 2->47976: 10.20.146.233:34060 -> 10.20.163.181:8080 tcp SYN
xx drop (Policy denied) flow 0x245f245f to endpoint 1158, identity 2->47976: 10.20.146.233:34060 -> 10.20.163.181:8080 tcp SYN
```

The cilium policy was applied to the pod A with IP address `10.20.163.181` and pod B was trying to talk to pod A. Listing CEPs for both pod A and B shows that they both had the correct ip addresses. However,  cilium seemed to think that the traffic coming from pod B is coming from ip `10.20.146.233` and was dropping it. The ip address `10.20.146.233` belonged to pod B before it was evicted from its host and deployed to another host earlier (we use spot instances for this cluster). So it looks like something in the cilium backend wasn't updated with the new ip address for pod B. 

We were able to fix this issue by manually restarting pod B. 

Note: Both pod A and B were being managed by cilium before and after being evicted from the host and getting deployed on another host. 

I attached  [cilium-sysdump](https://github.com/cilium/cilium/files/5692631/cilium-sysdump-20201214-134457.zip) for the source and destination nodes. Let me know if there is more info needed. Thank you in advance. 

**General Information**

- Cilium version (1.9.0)
- Kernel version (5.4.74-36.135.amzn2.x86_64)
- Orchestration system version in use (v1.18.9-eks-d1db3c)
- Generate and upload a system zip:
 [cilium-sysdump](https://github.com/cilium/cilium/files/5692631/cilium-sysdump-20201214-134457.zip)

**Comment by aanm** [comment]
@YesemKebede what is right IP address of pod B?

**Comment by YesemKebede** [comment]
@aanm the Ip address for pod B was `10.20.149.191` from `kubectl get CEP` but the drops showed `10.20.146.233` as its ip address. But pod A IP was `10.20.163.181`  just double checked. Uploded a new [sysdump](https://github.com/cilium/cilium/files/5696580/cilium-sysdump-20201215-094146.zip) incase If I made a mistake on the first one. Also we restarted pod B before capturing the sysdump so if the sysdump is taking snapshot it might not have the old IP. 

Thank you for taking a look.
